tests/bluetooth/mesh/src/microbit.c

```diff
@@ -33,10 +33,13 @@ static void configure_button(void)
 	gpio = device_get_binding(DT_ALIAS_SW0_GPIOS_CONTROLLER);
 
 	gpio_pin_configure(gpio, DT_ALIAS_SW0_GPIOS_PIN,
-			   (GPIO_DIR_IN | GPIO_INT | GPIO_INT_EDGE |
-			    GPIO_INT_ACTIVE_LOW));
+			   DT_ALIAS_SW0_GPIOS_FLAGS | GPIO_INPUT);
 
-	gpio_init_callback(&button_cb, button_pressed, BIT(DT_ALIAS_SW0_GPIOS_PIN));
+	gpio_init_callback(&button_cb, button_pressed,
+			   BIT(DT_ALIAS_SW0_GPIOS_PIN));
+
+	gpio_pin_interrupt_configure(gpio, DT_ALIAS_SW0_GPIOS_PIN,
```

Comment:
I had to add this here in order for the button to work, although the original code did not have a `gpio_pin_enable_callback()` here.

Reply:
Previously `gpio_pin_configure` configured and enabled the interrupts. It might be easier to read the code if we put a call to `gpio_pin_interrupt_configure` before `gpio_init_callback` or after `gpio_add_callback`. Both approaches should work fine.